homeassistant/auth/__init__.py

```diff
@@ -254,10 +260,47 @@ def get_auth_mfa_module(self, module_id: str) \
                 'System generated users cannot have refresh tokens connected '
                 'to a client.')
 
-        if not user.system_generated and client_id is None:
+        if token_type is None:
+            if user.system_generated:
+                token_type = models.TOKEN_TYPE_SYSTEM
+            else:
+                token_type = models.TOKEN_TYPE_NORMAL
+
+        if user.system_generated and token_type != models.TOKEN_TYPE_SYSTEM:
+            raise ValueError(
+                'System generated users can only have system type '
+                'refresh tokens')
+
+        if not user.system_generated and \
+                token_type == models.TOKEN_TYPE_SYSTEM:
+            raise ValueError(
+                'Only system generated users can have system type '
+                'refresh tokens')
+
+        if token_type == models.TOKEN_TYPE_NORMAL and client_id is None:
             raise ValueError('Client is required to generate a refresh token.')
 
-        return await self._store.async_create_refresh_token(user, client_id)
+        if (token_type == models.TOKEN_TYPE_LONG_LIVED_ACCESS_TOKEN and
+                client_name is None):
+            raise ValueError('Client_name is required for long-lived access '
+                             'token')
+
+        if token_type == models.TOKEN_TYPE_LONG_LIVED_ACCESS_TOKEN:
+            for token in list(user.refresh_tokens.values()):
+                if (token.client_name == client_name and token.token_type ==
```

Comment:
We should remove this check. There is no need for it.  If a user wants to use duplicate names, it's up to them.

Reply:
No check for 1:1 on client and refresh_token? Then you will list all refresh tokens on the UI?

Reply:
I want to add 2 lists to the UI per user, based on type. One list for all normal refresh tokens, one for all long lived access tokens.

```
Refresh Tokens:
 - example.com, last accessed 5 minutes ago
 - hassio.local:8123, last accessed 30 munutes ago

Access Tokens:
 - For my CURL script
 - For my other CURL script
```